library/include/rocwmma/internal/layout.hpp

```diff
@@ -204,6 +204,13 @@ namespace rocwmma
                 // ordering is broken.
                 static_assert(!(std::is_same<DataLayout, col_major>::value && VectorWidth > 1),
                               "ColNT in col_major does not support VectorWidth > 1");
+
+                // Must ensure that MaxVectorWidth fits inside the leading dimension
+                static_assert(
+                    (std::is_same<DataLayout, col_major>::value && (MaxVectorWidth <= BlockDim))
+                        || (std::is_same<DataLayout, row_major>::value
+                            && (MaxVectorWidth <= BlockK)),
```

Comment:
Due to the ColOrthoVW layout in row major, the VW mapping is in the BlockK dimension. If VW > BlockK then we risk OOB failures. More risky with smaller types because they will have higher VW.

As far as the MaxVectorWidth <= BlockDim  in col major this may be redundant.
Maybe the check should be generally MaxVW <= BlockDim